src/Sylius/Bundle/ApiBundle/Serializer/ProductNormalizer.php

```diff
@@ -54,6 +59,18 @@ public function supportsNormalization($data, $format = null, $context = []): boo
             return false;
         }
 
-        return $data instanceof ProductInterface;
+        return $data instanceof ProductInterface && $this->isShopGetOperation($context);
+    }
+
+    private function isShopGetOperation(array $context): bool
```

Comment:
We don't have any other operation for product then "get", but I would vote that this check should look for just get prefix - as if anything else pop up, it still should restrict disabled products in the shop
```suggestion
    private function isShopOperation(array $context): bool
```